### DISCUSSION TITLE
TIME fields compared without DATE part, feature or bug?

### DISCUSSION THREAD

**Discussion by TomasMoskala** [prisma/prisma#4630] [Q&A] [answered]
Hi,
I am using time fields in MariaDB

```sql
CREATE TABLE `table_with_time_field` (
  `table_with_time_field_id` INT NULL AUTO_INCREMENT,
  `time_field` TIME NOT NULL,
PRIMARY KEY (table_with_time_field_id`)) ENGINE=InnoDB DEFAULT CHARSET=utf8;
```
The data look they do not contain any date part
![image](https://user-images.githubusercontent.com/35471478/102072785-a5f3b100-3e02-11eb-978c-ae2a67c426e1.png)

If I compare like this 
```ts
ctx.prisma.table_with_time_field.findMany({where: {
      time_field: {gt: new Date()}},
});

```
I get results by comparing only the time part although reading the time field gives the date part included
e.g. "1970-01-01T17:00:00.000Z"

In this case if current timestamp is "2020-12-12T12:00:00.000Z" the function returns the row as if 
 "1970-01-01T17:00:00.000Z" > "2020-12-12T12:00:00.000Z"

I wonder if this is going to stay this way as I am using a lot of date comparisons in my project.

**Comment by nikolasburk** [comment, marked as answer]
Hey @TomasMoskala, ~this sounds like a bug to me. The comparisons for the `DateTime` type should include the the date part. Do you mind [creating a bug report](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=bug_report.md&title=) for this so that our engineers can pick this up?~  

EDIT: This is indeed by design, see comment below.

**Reply by TomasMoskala** [reply]
Are you sure it was not an intentional feature? It makes sense to compare just the time part if the field is of type TIME

**Reply by nikolasburk** [reply]
Yeah, let me check back internally how this is supposed to work. I assume this field is of type `DateTime` in your Prisma schema though, right? I guess the behaviour for that type should be consistent 🤔

**Reply by nikolasburk** [reply]
You were right indeed! Here's a the comment from one of our engineers explaining the behaviour in more detail:

It doesn't matter what the date part contains, if the database doesn't contain any date representation it can't compare the date part, because there's nothing to compare. DateTime is merely a superset serialization format for Time. Put differently:
- Sometimes the database allows you to write whatever you want on the protocol level, maybe even a full date, but will only store the relevant part for the storage type.
- Sometimes the drivers already handle the conversion to the expected storage type, losing precision or entire parts of the superset type used in the programming language.
- The same logically happens with any data used for comparisons. Type coercions or similar.


We serialize the superset type DateTime because as you saw correctly, we don't have Time. What you see in the data above is the zero value (1970) for the date part serialized / coerced at some point.

**Comment by TomasMoskala** [comment]
Great, thanks.
I hope it stays this way forever :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
